components/is-analytics/org.wso2.analytics.solutions.is.analytics/widgets/IsAnalyticsAverageSessionDuration/src/IsAnalyticsAverageSessionDuration.jsx

```diff
@@ -123,12 +124,12 @@ class IsAnalyticsAverageSessionDuration extends Widget {
 
     assembleQuery() {
         super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
-        let dataProviderConfigs = _.cloneDeep(this.state.providerConfig);
-        let query = dataProviderConfigs.configs.config.queryData.query;
+        const dataProviderConfigs = _.cloneDeep(this.state.providerConfig);
```

Comment:
I don't see `providerConfig` is being set to the `state` in the constructor. If you are putting something to the `this.state`, then it should be set in the constructor. (if you don't have a value then set a default value or `null`)

Reply:
@Ambujaa10 Is this been addressed?

Reply:
yes. I assigned providerConfig to null in this.state